examples/nginx-app/Dockerfile

```diff
@@ -1,35 +1,50 @@
 # nginx-centos7
 # Here you can use whatever image base is relevant for your application.
-FROM centos:centos7
+FROM openshift/base-centos7
 
 # Here you can specify the maintainer for the image that you're building
 MAINTAINER Victor Palade <ipalade@redhat.com>
 
 # Export the environment variable that provides information about the application.
 # Replace this with the according version for your application.
-ENV NGINX_VERSION=1.6.3
+ENV NGINX_VERSION=1.10.1
 
 # Set the labels that are used for OpenShift to describe the builder image.
 LABEL io.k8s.description="Nginx Webserver" \
-    io.k8s.display-name="Nginx 1.6.3" \
+    io.k8s.display-name="Nginx 1.10.1" \
     io.openshift.expose-services="8080:http" \
     io.openshift.tags="builder,webserver,html,nginx" \
-    # this label tells s2i where to find its mandatory scripts
-    # (run, assemble, save-artifacts)
-    io.openshift.s2i.scripts-url="image:///usr/libexec/s2i"
+    authors="Victor Palade <ipalade@redhat.com>,Luis Fernando Gommes <dev@luiscoms.com.br>"
 
 # Install our Nginx package and clean the yum cache so that we don't have any
 # cached files waste space.
 RUN yum install -y epel-release  && \
     yum install -y --setopt=tsflags=nodocs nginx && \
     yum clean all
 
+# Create dir to mantain pidfile
+RUN mkdir -p /opt/app-root/run/
+
 # We will change the default port for nginx (It's required if you plan on
 # running images as non-root user).
-RUN sed -i 's/80/8080/' /etc/nginx/nginx.conf
+RUN sed -i 's|pid /run/nginx.pid;|pid /opt/app-root/run/nginx.pid;|; \
+         s|user nginx;||; \
+         s|error_log.*|error_log /dev/stderr;|; \
+         s|access_log.*|access_log /dev/stdout main;|; \
+         s|/usr/share/nginx/html|/opt/app-root/src|; \
+         s|80|8080|;' /etc/nginx/nginx.conf
+
+# Copy the S2I scripts to /usr/libexec/s2i, since openshift/base-centos7 image sets io.openshift.s2i.scripts-url label that way, or update that label
+COPY ./.s2i/bin/ $STI_SCRIPTS_PATH
+
+# Drop the root user and make the content of /opt/app-root owned by user nginx
+RUN chown -R nginx:nginx /opt/app-root
+
+# Run container as nginx user
+USER nginx
```

Comment:
everything that is below this point should actually take into account this comment: https://github.com/openshift/source-to-image/pull/371#issuecomment-160782599

the Makefile, Run as user X, etc these weren't things that we wanted to show the user at first.
The idea of the tutorial was to make it as easy as possible to get up and running with s2i with the bare minimum. I know you spent some time to get this to work, but i don't think it's what we're looking for.
@bparees what do you think, should we accept this customization for the tutorial?